### ISSUE TITLE
What's the difference between 'binary' and 'xentropy' as the objective?

### ISSUE THREAD

**Issue by bbennett36** [Microsoft/LightGBM#1348]
I'm confused about these two objective functions. The documentation says that the 'binary' objective is cross entropy but when I use 'xentropy', I get different results. It looks like it is always performing slightly better using 'xentropy' instead of 'binary'. I'm wondering why the results are different and if I'm interpreting these parameters the wrong way. The 'binary' parameter has a link to the cross entropy wiki page so I'm just not sure what's going on here. If there is a difference between the two, could someone explain further for me? I do know that Cross Entropy can be used for multiclass problems but this is for a binary classification problem.

early_stopping_rounds = 100
learning_rate = 0.01
evals_result = {} 

params = {
        'application': 'binary',
        'learning_rate': learning_rate,
        'num_leaves': 40,
        'feature_fraction': 1.0,
        'max_bin': 63,
        'num_threads': 16, 
        'metric': ['auc']
}
        

lgb_model = lgb.train(params, lgb_train, early_stopping_rounds, valid_sets=[lgb_test], evals_result=evals_result,  verbose_eval=100)

`[100]	valid_0's auc: 0.993791`

early_stopping_rounds = 100
learning_rate = 0.01
evals_result = {} 

params = {
        'application': 'xentropy',
        'learning_rate': learning_rate,
        'num_leaves': 40,
        'feature_fraction': 1.0,
        'max_bin': 63,
        'num_threads': 16, 
        'metric': ['auc']
}

lgb_model = lgb.train(params, lgb_train, early_stopping_rounds, valid_sets=[lgb_test], evals_result=evals_result,  verbose_eval=100)

`[100]	valid_0's auc: 0.995936`

Thank you!

**Comment by bbennett36** [comment]
@Laurae2 Hmmmm so what would cause different results for a classification problem if there is only labels for 1 and 0?

**Comment by Laurae2** [comment]
@bbennett36 probably different initialization score. I think it was not yet implemented in `xentropy`.

ping @guolinke for more information

**Comment by guolinke** [comment]
yeah, exactly, xentropy will start from the average probability: https://github.com/Microsoft/LightGBM/blob/master/src/objective/xentropy_objective.hpp#L108-L128
